middleware/logging.go

```diff
@@ -25,13 +27,22 @@ func (l Log) Wrap(next http.Handler) http.Handler {
 		if l.LogSuccess || !(100 <= i.statusCode && i.statusCode < 400) {
 			log.Infof("%s %s (%d) %s", r.Method, uri, i.statusCode, time.Since(begin))
 		}
+		if l.LogRequestHeaders {
+			headers, err := httputil.DumpRequest(r, false)
```

Comment:
May want to consider doing this before the call to `next.ServeHTTP`, as some handlers modify the request.